controllers/twophase/types.go

```diff
@@ -61,206 +61,80 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	}
 	chaos := _chaos.(v1alpha1.InnerSchedulerObject)
 
-	duration, err := chaos.GetDuration()
-	if err != nil {
-		r.Log.Error(err, "failed to get chaos duration")
-		return ctrl.Result{}, err
+	status := chaos.GetStatus()
+
+	targetPhase := status.Experiment.Phase
+
+	if !chaos.GetNextRecover().IsZero() && chaos.GetNextRecover().Before(now) {
+		targetPhase = v1alpha1.ExperimentPhaseWaiting
 	}
 
-	scheduler := chaos.GetScheduler()
-	if scheduler == nil {
-		r.Log.Info("Scheduler should be defined currently")
-		return ctrl.Result{}, fmt.Errorf("misdefined scheduler")
+	if chaos.GetNextStart().Before(now) {
+		targetPhase = v1alpha1.ExperimentPhaseRunning
```

Comment:
The logic here seems not correct, if we use the cron that does not start right away, for example, if I want to start a chaos at 00:00 each day for 10 minutes. I still think a `Previous` function, to get `lastStart` and `lastRecover` could be useful in this case. But since it's not the purpose of this PR, we could add this in another PR

Reply:
Yes. It's still a problem 🤦‍♂️ .

Reply:
@Yiyiyimu This is a bug here, Can you help to create an issue about this bug?

Reply:
@cwen0 Sure, created as #1270